csharp/test/Apache.Arrow.Tests/Apache.Arrow.Tests.csproj

```diff
@@ -3,6 +3,9 @@
 
   <PropertyGroup>
     <TargetFrameworks>net7.0</TargetFrameworks>
+    <!-- Unloading the AppDomain causes net472 tests to hang on completion when Python has been used
+    <TargetFrameworks>net7.0;net472</TargetFrameworks>
+    -->
```

Comment:
Any idea why?

Reply:
It looks like we need to call PythonEngine.Shutdown.

Reply:
Do you want to make that fix as part of this PR?  Or would you rather that get handled in a follow-up?

Reply:
Yeah, I figured it out.

Reply:
I ended up reverting the test project change because it breaks CI, and I can't figure out how to run 472 only on Windows. I filed #36815 to track this gap.